rest-api/src/test/java/com/cloudbees/workflow/rest/endpoints/JobAndRunAPITest.java

```diff
@@ -95,6 +150,58 @@ public void test() throws Exception {
         assertSinceQueryParamOkay(job, webClient);
     }
 
+    @Test
+    public void testErrorHandling() throws Exception {
+        WorkflowJob job = jenkinsRule.jenkins.createProject(WorkflowJob.class, "Failing Job");
+
+        job.setDefinition(new CpsFlowDefinition("" +
+                "stage \"pass\"\n" +
+                "echo \"I'm passing\"\n" +
+                "\n" +
+                "stage \"catch-an-error\"\n" +
+                "catchError {\n" +
+                "   error \"You won't see me?\"\n" +
+                "}\n" +
+                "\n" +
+                "stage \"and now keep passing\"\n" +
+                "echo \"still passing\""));
+
+        QueueTaskFuture<WorkflowRun> build = job.scheduleBuild2(0);
+        jenkinsRule.assertBuildStatus(Result.FAILURE, build.get());
+
+        JenkinsRule.WebClient webClient = jenkinsRule.createWebClient();
+
+        assertBasicJobInfoOkay(job, webClient);
+
+        String jobRunsUrl = job.getUrl() + "wfapi/runs?fullStages=true";
+        Page runsPage = webClient.goTo(jobRunsUrl, "application/json");
+        String jsonResponse = runsPage.getWebResponse().getContentAsString();
+        JSONReadWrite jsonReadWrite = new JSONReadWrite();
+        RunExt[] workflowRuns = jsonReadWrite.fromString(jsonResponse, RunExt[].class);
+
+        Assert.assertEquals(1, workflowRuns.length);
+        RunExt runExt = workflowRuns[0];
+        Assert.assertEquals(StatusExt.FAILED, runExt.getStatus());
+        Assert.assertEquals(3, runExt.getStages().size());
+
+        for (StageNodeExt st : runExt.getStages()) {
+            assertStageInfoOkay(st, true);
+        }
+
+        Assert.assertEquals(StatusExt.SUCCESS, runExt.getStages().get(0).getStatus());
+        Assert.assertEquals(StatusExt.SUCCESS, runExt.getStages().get(1).getStatus());
+        Assert.assertEquals(StatusExt.FAILED, runExt.getStages().get(2).getStatus());
+
+        StageNodeExt finalStage = runExt.getStages().get(2);
+        AtomFlowNodeExt finalNode = finalStage.getStageFlowNodes().get(finalStage.getStageFlowNodes().size()-1);
+        Assert.assertEquals(StatusExt.FAILED, finalNode.getStatus());
+        Assert.assertNotNull(finalNode.getError());
+        Assert.assertNotNull(finalNode.getError().getMessage());
```

Comment:
If you localize the placeholder error message using `Messages` you could `assertEquals` here.